### PR TITLE
Gutenberg: Simple payments basic edit/save

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

* Add simple edit that can be used to provide an ID of existing simple payments
* Add save that will produce simple payments shortcode
* Update supports block to better define the block
  https://wordpress.org/gutenberg/handbook/block-api/#supports-optional

#### Testing instructions

* https://calypso.live/?branch=update/simple-payments-save-payment-id&flags=gutenberg/block/simple-payments
* Add a simple payment (`/paym`)
* Input some number
* Check rendered HTML to see correct shortcode
* Rendered shortcode on frontend should appear correctly if simple payment exists

### CODE REVIEW

File: client/gutenberg/extensions/simple-payments/editor.js [hunk 1]
```diff
@@ -22,16 +28,18 @@ registerBlockType( 'jetpack/simple-payments', {
...
-			This block is under development and not ready for production.
-		</div>
-	),
+	edit,
+
+	save,
 
-	save: () => null,
+	supports: {
+		className: false,
```
Comment (sirreal) [inline client/gutenberg/extensions/simple-payments/editor.js]: Don't add Gutenberg auto-classname

File: client/gutenberg/extensions/simple-payments/editor.js [hunk 2]
```diff
@@ -22,16 +28,18 @@ registerBlockType( 'jetpack/simple-payments', {
...
+		customClassName: false,
```
Comment (sirreal) [inline client/gutenberg/extensions/simple-payments/editor.js]: Don't add custom classname to sidebar

File: client/gutenberg/extensions/simple-payments/editor.js [hunk 3]
```diff
@@ -22,16 +28,18 @@ registerBlockType( 'jetpack/simple-payments', {
...
+		html: false,
```
Comment (sirreal) [inline client/gutenberg/extensions/simple-payments/editor.js]: Don't allow HTML editing of the block

**Comment by simison** [comment]
The output is now:

```html
<!-- wp:jetpack/simple-payments {"paymentId":333} -->
[simple-payment id="333"]
<!-- /wp:jetpack/simple-payments -->
```

There probably isn't any other way to keep `paymentId` away from block meta than storing it only in the state? 

We would catch the ID just fine with shortcode `transform` from a block like this:
```
<!-- wp:jetpack/simple-payments -->
[simple-payment id="333"]
<!-- /wp:jetpack/simple-payments -->
```

File: client/gutenberg/extensions/simple-payments/edit.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		}
+	};
+
+	render() {
+		const { attributes, instanceId } = this.props;
+		const inputId = `payment-id-${ instanceId }`;
+		return (
+			<Fragment>
+				<label htmlFor={ inputId }>Simple Payment ID</label>
+				<input
```
Comment (simison) [inline client/gutenberg/extensions/simple-payments/edit.js]: Throws:
> Warning: A component is changing an uncontrolled input of type number to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component.

Comment (sirreal) [reply client/gutenberg/extensions/simple-payments/edit.js]: I've just pushed a fix for this.

This is just getting something into place, warnings shouldn't be too concerning at this point.

**Comment by sirreal** [comment]
> There probably isn't any other way to keep `paymentId` away from block meta than storing it only in the state?

Yep, the duplication is bothersome. Let's explore in follow ups 👍

File: client/gutenberg/extensions/simple-payments/save.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { RawHTML } from '@wordpress/element';
+
+export default function Save( { attributes } ) {
+	const { paymentId } = attributes;
+	return paymentId ? <RawHTML>{ `[simple-payment id="${ paymentId }"]` }</RawHTML> : null;
```
Comment (simison) [inline client/gutenberg/extensions/simple-payments/save.js]: This seems to work without `RawHTML` just fine:
```js
return paymentId ? `[simple-payment id="${ paymentId }"]` : null;
```

Any reason to keep it?

Comment (sirreal) [reply client/gutenberg/extensions/simple-payments/save.js]: 🔪 c44a37cfb4f3b40f074fc9fdb72648d7c49112eb

**Pull request merged**